streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -324,6 +324,31 @@ private void closeDirtyAndRevive(final Collection<Task> taskWithChangelogs, fina
         }
     }
 
+    private Map<Task, Set<TopicPartition>> assignStartupTasks(final Map<TaskId, Set<TopicPartition>> tasksToAssign,
+                                                              final String threadLogPrefix,
+                                                              final TopologyMetadata topologyMetadata,
+                                                              final ChangelogRegister changelogReader) {
+        if (stateDirectory.hasStartupTasks()) {
+            final Map<Task, Set<TopicPartition>> assignedTasks = new HashMap<>(tasksToAssign.size());
+            for (final Map.Entry<TaskId, Set<TopicPartition>> entry : tasksToAssign.entrySet()) {
+                final TaskId taskId = entry.getKey();
+                final Task task = stateDirectory.removeStartupTask(taskId);
+                if (task != null) {
+                    // replace our dummy values with the real ones, now we know our thread and assignment
+                    final Set<TopicPartition> inputPartitions = entry.getValue();
+                    task.stateManager().assignToStreamThread(new LogContext(threadLogPrefix), changelogReader, inputPartitions);
+                    task.updateInputPartitions(inputPartitions, topologyMetadata.nodeToSourceTopics(taskId));
```

Comment:
What happens if the task has topics from intermediate internal topics? Since `topologyMetadata` doesn't return topics with a prefix, it could cause a topology exception when recycling a standby task to an active task.